packages/dds/tree/src/simple-tree/api/schemaFactoryRecursive.ts

```diff
@@ -283,6 +283,6 @@ export function allowUnused<T>(t?: T): void {}
  * }
```

Comment:
Probably not for this PR (as its alpha), but what do you think about moving toward the allowUnused pattern to work with more typescript? configs?

```typescript
* allowUnused<ValidateRecursiveSchema<typeof RecursiveArray>>();
```